ui-v2/src/hooks/variables.ts

```diff
@@ -11,82 +11,91 @@ import {
- *  - placeholderData: Uses previous data while loading new data
- */
-const buildVariablesQuery = (options: UseVariablesOptions) =>
-	queryOptions({
-		queryKey: variableKeys.filtered(options),
-		queryFn: async () => {
-			const response = await getQueryService().POST("/variables/filter", {
-				body: options,
-			});
-			return response.data;
-		},
-		staleTime: 1000,
-		placeholderData: keepPreviousData,
+const deleteVariableRequest = (id: string) => {
+	return getQueryService().DELETE("/variables/{id}", {
+		params: { path: { id } },
 	});
+};
+
+// ----- 🔑 Queries 🗄️
+// ----------------------------
 
 /**
- * Builds a query configuration for counting variables with optional filters
- * @param options - Optional filter options for the variables count query
- * @returns A query configuration object compatible with TanStack Query including:
- *  - queryKey: A unique key for caching the count results
- *  - queryFn: The async function to fetch the filtered count
- *  - staleTime: How long the data should be considered fresh (1 second)
- *  - placeholderData: Uses previous data while loading new data
+ * ```
+ *  🏗️ Variable queries construction 👷
+ *  all   =>   ['variables'] // key to match ['variables', ..., ...]
+ *  list  =>   ['variables', 'list'] // key to match ['variables', 'list', ..., ...]
+ *             ['variables', 'list', { ...withOptionsFoo }]
+ *             ['variables', 'list', { ...withOptionsBar }]
+ *             ['variables', 'list', { ...withOptionsBaz }]
+ *  count =>   ['variables', 'count'] // key to match ['variables', 'count', ..., ...]
+ *             ['variables', 'count', {}] // all count
+ *             ['variables', 'count', { ...withOptionsFoo }]
+ *             ['variables', 'count', { ...withOptionsBar }]
+ *             ['variables', 'count', { ...withOptionsBaz }]
+ * ```
  */
-const buildCountQuery = (options?: UseVariablesOptions) =>
-	queryOptions({
-		queryKey: variableKeys.filteredCount(options),
-		queryFn: async () => {
-			const body = options?.variables ? { variables: options.variables } : {};
-			const response = await getQueryService().POST("/variables/count", {
-				body,
-			});
-			return response.data;
-		},
-		staleTime: 1000,
-		placeholderData: keepPreviousData,
-	});
+const variableQueries = {
+	all: () => ["variables"] as const,
+	lists: () => [...variableQueries.all(), "list"] as const,
+	list: (options: UseVariablesOptions) =>
+		queryOptions({
+			queryKey: [...variableQueries.lists(), options] as const,
+			queryFn: () => fetchVariablesRequest(options),
+			staleTime: 1000,
+			placeholderData: keepPreviousData,
+		}),
+	counts: () => [...variableQueries.all(), "count"] as const,
+	count: (variables?: Variables) => {
+		const body = { variables };
+		return queryOptions({
+			queryKey: [...variableQueries.counts(), body] as const,
+			queryFn: () => fetchVariableCountRequest(body),
+			staleTime: 1000,
+			placeholderData: keepPreviousData,
+		});
```

Comment:
We want to ensure that we aren't including the body in the query key `variables` is undefined to avoid making duplicate variable count calls when the variables page first mounts.